spring-integration-redis/src/main/java/org/springframework/integration/redis/outbound/RedisStoreWritingMessageHandler.java

```diff
@@ -284,49 +277,48 @@ private void writeToZset(RedisZSet<Object> zset, final Message<?> message) throw
 		final Object payload = message.getPayload();
 		final BoundZSetOperations<String, Object> ops =
 				(BoundZSetOperations<String, Object>) this.redisTemplate.boundZSetOps(zset.getKey());
-		final boolean zsetIncrementHeader = this.extractZsetIncrementHeader(message);
+		boolean zsetIncrementHeader = extractZsetIncrementHeader(message);
 		if (this.extractPayloadElements) {
 			if ((payload instanceof Map<?, ?> && this.verifyAllMapValuesOfTypeNumber((Map<?, ?>) payload))) {
-				final Map<Object, Number> payloadAsMap = (Map<Object, Number>) payload;
-				this.processInPipeline(() -> {
+				Map<Object, Number> payloadAsMap = (Map<Object, Number>) payload;
+				processInPipeline(() -> {
 					for (Entry<Object, Number> entry : payloadAsMap.entrySet()) {
 						Number d = entry.getValue();
 						incrementOrOverwrite(ops, entry.getKey(), d == null ?
-								determineScore(message) :
-								NumberUtils.convertNumberToTargetClass(d, Double.class),
+										determineScore(message) :
+										NumberUtils.convertNumberToTargetClass(d, Double.class),
 								zsetIncrementHeader);
 					}
 				});
 			}
 			else if (payload instanceof Collection<?>) {
-				this.processInPipeline(() -> {
+				processInPipeline(() -> {
 					for (Object object : ((Collection<?>) payload)) {
 						incrementOrOverwrite(ops, object, determineScore(message), zsetIncrementHeader);
 					}
 				});
 			}
 			else {
-				this.incrementOrOverwrite(ops, payload, this.determineScore(message), zsetIncrementHeader);
+				incrementOrOverwrite(ops, payload, this.determineScore(message), zsetIncrementHeader);
 			}
 		}
 		else {
-			this.incrementOrOverwrite(ops, payload, this.determineScore(message), zsetIncrementHeader);
+			incrementOrOverwrite(ops, payload, this.determineScore(message), zsetIncrementHeader);
 		}
 	}
 
 	private boolean extractZsetIncrementHeader(Message<?> message) {
 		if (message.getHeaders().containsKey(RedisHeaders.ZSET_INCREMENT_SCORE)) {
```

Comment:
If we open up the expression for user specification, this if test would have to go inside the default `FunctionExpression`.